File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
import math

def solution(n, words):
    answer = []
    talk = []
    for i in range(len(words)) : 
        if i == 0 :
            talk.append(words[i])
            continue
        if (words[i] not in talk) and words[i][0] == words[i-1][-1] :
            talk.append(words[i])
            continue
        else :
            #talk에 있거나 끝말잇기가 안되었거나 
            answer.append((i%n)+1)
            answer.append((i//n)+1)
            break
    
    if len(answer) == 0 :
        answer.append(0)
        answer.append(0)
        
    return answer
            
    

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = {0,0};

        Set<String> wordset = new HashSet<>();
        
        char beforechar = words[0].charAt(words[0].length()-1);
        wordset.add(words[0]);
        
        for (int i = 1; i<words.length; i++)
        {
            String now = words[i];
            
            if (now.charAt(0) != beforechar)
            {
                answer[0] = (i+1)%n == 0 ? n : (i+1)%n;
                answer[1] = (i+1)%n == 0 ? ((i+1)/n) : ((i+1)/n)+1;
                return answer;
            }
            
            if (wordset.contains(now))
            {
                answer[0] = (i+1)%n == 0 ? n : (i+1)%n;
                answer[1] = (i+1)%n == 0 ? ((i+1)/n) : ((i+1)/n)+1;
                return answer;
            }
            
            wordset.add(now);
            
            beforechar = now.charAt(now.length()-1); 
        }

        return answer;
    }
}